### DISCUSSION TITLE
Bee modelling

### DISCUSSION THREAD

**Discussion by Paculino** [mars-sim/mars-sim#669] [Ideas]
How should bee modelling be expanded?

### Risks to bees
The longer day-night cycle should not pose any risk, since their circadian rhythm is already quite flexible, and non-existent when larval ([Circadian Rhythms and Sleep in Honey Bees](https://link.springer.com/chapter/10.1007/978-94-007-2099-2_3)). However, while large greenhouses are fine for bees, smaller ones may be insufficient. Most diseases may be kept out, although some ailments are not the result of fungi, bacteria, or viruses. Time spent in ambient conditions below 10 °C is proportional to risk of dysentery according to [Wikipedia](https://en.wikipedia.org/wiki/List_of_diseases_of_the_honey_bee). Colony Collapse Disorder is poorly understood, and harmful fungi may survive despite precautions.
### Risks to settlement
Other issues, while not a risk to the bees, could endanger or inconvenience humans in the settlement. A settlement could find itself without bees at least until the next resupply. If a settlement is without bees (or has too few bees), certain crops would require manual pollination (or avoidance). Stings may be more likely since the smoke typically used to discourage aggression would not be advisable. The bees may grow in population more than expected and start building a second hive anywhere they can reach; would multiple hives result in worse health for each, since there is no increase in food?
### Questions
Should one hive serve only one greenhouse, or should the bees be able to go through halls and serve two greenhouses? Would bees get lost and die if they can go through the halls, since neither the sun nor the hive would be visible? If bees are not meant to be able to leave the greenhouse of their hive, it nevertheless seems unlikely that they would be kept there perfectly as people go in and out; would this significantly affect the bee population? How long would manual pollination take? Ten seconds per bloom seems reasonable, but people may miss blooms. For crops which can self-pollinate such as tomatoes, should there be any benefit to also having bees available as pollinators? Should honey have grades, since it can be bitter if resulting from certain crops? A manufacturing process is needed to be able to create a hive box, frames, and possibly a small transport cage for the queen? What should the box/frames be made of? Solid wax may confuse the bees. Metal could simply be heated (even by conduction) to easily remove the wax by melting it, but plastic may be easier.

**Comment by bevans2000** [comment]
I like the idea of having BeeKeeping in the simulation. We would model it as another ```Function```. 
The question is to what level of modelling do we go to? Simulating individual Bees would be too low. The approach with the Fishery function works well of just simulating the population size.

**Reply by mokun** [reply]
> BeeKeeping ... model it as another `Function`. 

Yes, let's create a new `FunctionType` called BEEKEEPING and its associated classes.

> The question is to what level of modelling do we go to? Simulating individual Bees would be too low. 

We can start modeling it macroscopically. We can also set up an activity index for each class of bees (worker bee/drone, queen, larvae, pupae, etc.) 

> The approach with the Fishery function works well of just simulating the population size.

Yes and in terms of fishery, we should advertise how many fishes have been farmed/harvested (preferably toward the end of their lifecycles) and get an achievement board for comparison.

A quality-of-life index could be created for living each settlement. Maintaining a garden, farming fishes and growing bees would certainly improve this index a bit.

**Comment by mokun** [comment]
> The longer day-night cycle should not pose any risk, since their circadian rhythm is already quite flexible, and non-existent when larval ([Circadian Rhythms and Sleep in Honey Bees](https://link.springer.com/chapter/10.1007/978-94-007-2099-2_3)).

Interesting that you point out bees's circadian clock.  

I suppose when there is enough illumination (sunlight plus artificial light), their level of activities would be higher and longer.

> large greenhouses are fine for bees, smaller ones may be insufficient.

Right. We can first aim at growing bees at Large Greenhouse building.

But in future we could link those small greenhouses back-to-back into one continuous space with connecting tubes going between them.

**Comment by mokun** [comment]
> For crops which can self-pollinate such as tomatoes, should there be any benefit to also having bees available as pollinators?

Sure. We can create a pollination index that's moderated by the activity level of the bees.

Let say this pollination index will impact the flowering phase (which currently occupies 25% of the growing time by default) of a fruit crop.

Currently, we have a total of 5 fruit crops, namely, tomato, blueberry, cranberry, strawberry, and green bell pepper.

Note that a fruit crop is simply a crop with crop-category="fruits" defined in crop.xml.
